packages/react-components/react-table/src/utils/columnResizeUtils.ts

```diff
@@ -0,0 +1,188 @@
+import { TableColumnDefinition, ColumnWidthState, TableColumnId, TableColumnSizingOptions } from '../hooks';
+
+const DEFAULT_WIDTH = 150;
+const DEFAULT_MIN_WIDTH = 100;
+
+/**
+ * This function takes the column definitions and the curent ColumnWidthState and returns new state.
+ *  - It uses existing state for existing columns.
+ *  - It removes any state for columns no longer present.
+ *  - It checks if any column has been replaced and returns updated state if so
+ *  - It returns old state if no changes in the state have been made (so that react doesn't call effects)
+ * @param columns
+ * @param state
+ * @returns
+ */
+
+export function columnDefinitionsToState<T>(
+  columns: TableColumnDefinition<T>[],
+  state: ColumnWidthState[] = [],
+  columnSizingOptions: TableColumnSizingOptions = {},
+): ColumnWidthState[] {
+  let updated = false;
```

Comment:
we discussed offline that this `updated` flag might not be necessary, did you try to refactor with a simple reference check?

Reply:
Its easier to just mark updated immediately as we know something has changed, so that we can decide at the end if we need to return a new state or original state based on a boolean instead of having to go over both states and compare references 🤷